docs/source/Participate/PlatformIO.rst

```diff
@@ -176,10 +180,66 @@ All these mainly apply to the latest ESP32-xx builds using ESP-IDF5.x.
 On a beefy desktop PC, like an AMD Ryzen9 with 12 CPU cores, building a ``MAX`` build for ESP32-S3 takes about 10 - 15 minutes on Windows.
 The same build when using WSL2 via VS-Code takes only 2 minutes to build.
 
+When using VS-Code with WSL2, the VS-Code editor application runs on Windows, and, via the WSL plugin, communicates with a VS-Code server component on a Linux instance, running on WSL, that has the source code repository available, and where the build process is executed.
+
 
 Installing WSL2
 ---------------
 
+For installing WSL2 a couple of steps are needed. Assumption here is that the default Linux distribution for WSL is used, currently Ubuntu (24.04 LTS), using apt-get as the package manager. Other Linux distributions may use a different package manager, like dnf, zypper or yum, using somewhat different commands. Those commands are not documented here.
+
+Install WSL2 - The Windows part
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Open a Windows Command prompt, preferrably Powershell, with Administrator privileges (Press the Windows key and start typing ``powershell``, then select Run as administrator)
+
+* Type the command ``wsl --install`` and press <Enter>, this will start the installation procedure, ask for permissions via User Account Control where needed (that must be confirmed), and download and install what's required.
+
+* Once this part of the installation is complete, the computer must be restarted. This will install any additional Windows components that weren't installed yet.
+
+* After logging in to Windows, the installation procedure will be further completed, and you will be asked to enter a username and password for your day-to-day Linux user. This password will be needed later, f.e. when starting the WSL Linux instance, so best practice is to store it in a password safe of some sort (or write it on a Post-It note and stick that under your keyboard ;-)) 
+
+Install WSL2 - The Linux part
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that Linux is installed, it's strongly advised to get it up to date with the latest patches, similar to Windows Update. This should be repeated on a regular basis to keep your installation up to date:
+
+.. highlight::sh
+
+.. code-block::
+    
+    sudo apt-get update
+    sudo apt-get upgrade
+
+NB: The ``sudo`` command will ask for your current account password once, (and maybe later again after some time has passed). When asked for confirmation, confirm the installation of any needed updates, or add the ``-y`` parameter to the ``upgrade`` command to continue without questions.
+
+Additionally some tools need to be installed so PlatformIO can be properly installed later, and we can use the Uncrustify plugin in VS-Code to format the code nicely:
+
+.. highlight::sh
+
+.. code-block::
+    
+    sudo apt-get install python3-venv
+    sudo apt-get install python3-pip
+    sudo apt-get install uncrustify
```

Comment:
These 3 can also be put in the same command line.

+ add a N.B. about uncrustify when using Ubuntu 22.04 as the version supplied with Ubuntu 22.04 is _not_ working well.